pyroSAR/drivers.py

```diff
@@ -1758,62 +1758,50 @@ def resolution(self):
-                # waz = float(az_proc.find('windowCoefficient').text)
-                # baz = float(az_proc.find('processingBandwidth').text)
-                # kbaz = b_factors[coefficients.index(waz)]
-                # velocities = tree.findall('.//orbit/velocity')
-                # ids = ['x', 'y', 'z']
-                # xyz = [tuple(float(a.find(b).text) for b in ids) for a in velocities]
-                # velocities = [math.sqrt(sum([b ** 2 for b in a])) for a in xyz]
-                # vsat = mean(velocities)  # around 7590
-                # resolutions_az.append(0.886 * vsat / baz * kbaz)
-                if self.acquisition_mode == 'IW':
-                    resolutions_az.append(22.)
-                elif self.acquisition_mode == 'EW':
-                    resolutions_az.append(43.)
-                else:
-                    RuntimeError("acquisition mode '{}' not supported".format(self.acquisition_mode))
-            resolution_rg = median(resolutions_rg)
-            resolution_az = median(resolutions_az)
-        elif self.product == 'GRD':
-            match = re.match(re.compile(self.pattern), os.path.basename(self.file))
-            resolution_class = match.group('resolution')
-            resolution_rg = resolution_az = None
-            if resolution_class == 'F':
-                resolution_rg = 9
-                resolution_az = 9
-            elif resolution_class == 'H':
-                resolution_rg = {'SM': 23, 'IW': 20, 'EW': 50}[self.acquisition_mode]
-                resolution_az = {'SM': 23, 'IW': 22, 'EW': 50}[self.acquisition_mode]
-            elif resolution_class == 'M':
-                resolution_rg = {'SM': 84, 'IW': 88, 'EW': 93, 'WV': 52}[self.acquisition_mode]
-                resolution_az = {'SM': 84, 'IW': 87, 'EW': 87, 'WV': 51}[self.acquisition_mode]
-            else:
-                raise RuntimeError("unknown resolution class: {}".format(resolution_class))
-        else:
-            raise RuntimeError("unsupported product: {}".format(self.product))
-        self.meta['resolution'] = float(resolution_rg), float(resolution_az)
-        return float(resolution_rg), float(resolution_az)
+        resolutions_rg = []
+        resolutions_az = []
+        for i, par in enumerate(proc_pars):
+            # computation of slant range resolution
+            rg_proc = par.find('rangeProcessing')
+            wrg = float(rg_proc.find('windowCoefficient').text)
+            brg = float(rg_proc.find('processingBandwidth').text)
+            lbrg = float(rg_proc.find('lookBandwidth').text)
+            lrg = brg / lbrg
+            kbrg = b_factors[coefficients.index(wrg)]
+            resolutions_rg.append(0.886 * c / (2 * brg) * kbrg * lrg)
+            
+            # computation of azimuth resolution; yet to be checked for correctness
+            az_proc = par.find('azimuthProcessing')
+            laz = int(az_proc.find('numberOfLooks').text)
```

Comment:
Probably it makes sense to use the same formula used for RG, it should work also for GRD MR and LR.

Reply:
done